File: Behavioral/Strategy/CSharp/PaymentProcessor.cs
// PaymentProcessor.cs

using System;

public class PaymentProcessor {
  private IPaymentStrategy? _paymentStrategy;

  public void SetPaymentStrategy(IPaymentStrategy paymentStrategy) {
    _paymentStrategy = paymentStrategy;
  }

  public void ProcessPayment(double amount) {
    if (_paymentStrategy == null) {
      Console.WriteLine("No payment strategy selected.");
      return;
    }
    _paymentStrategy.Pay(amount);
  }
}

File: Behavioral/Strategy/Python/payment_processor.py
# payment_processor.py

class PaymentProcessor:
  def __init__(self):
    self._payment_strategy = None

  def set_payment_strategy(self, payment_strategy):
    self._payment_strategy = payment_strategy

  def process_payment(self, amount):
    if self._payment_strategy is None:
      print("No payment strategy selected.")
      return
    self._payment_strategy.pay(amount)
